src/Services/UploadImageService.php

```diff
@@ -223,14 +225,14 @@ protected function setUploadedImage(string $imageAttributeName)
         }
 
         // An image is already uploaded, a new one is uploaded
-        if (ends_with($value, '.jpg') && !empty($originalValue)) {
+        if (preg_match(self::R_MD5_MATCH, $value) && !empty($originalValue)) {
```

Comment:
You should never use the keyword `self` unless you have no choice (expression in a constant, PHP < 5.3) or you explicitly want to prevent it from being overloaded. Better use `static` instead.

Reply:
If you remove the check for `.jpg` extension, your PR also extends support to all image formats. You have to notice it or make another PR.

And maybe instead of `preg_match` you should take a look at http://php.net/pathinfo to check if the file has an extension and is therefore not a base64 encoded string, I think it would be more robust. Any thoughts on this @jguyomard ?

Reply:
+1